object/state.go

```diff
@@ -94,6 +96,43 @@ func (e *Environment) RegisterTrie(t *trie.Trie) {
 	}
 }
 
+// Returns the number of ids written.
+func (e *Environment) SaveGlobals(to io.Writer) (int, error) {
+	for e.outer != nil {
+		e = e.outer
+	}
+	keys := make([]string, 0, len(e.store))
+	for k := range e.store {
+		keys = append(keys, k)
+	}
+	slices.Sort(keys)
+	n := 0
+	for _, k := range keys {
```

Comment:
I might be missing something but either you add a comment explaining why you need `n`

or you do that


```suggestion
	for n, k := range keys {
```

Then return n+1 at the end of the function

Reply:
n would be out of scope...

Reply:
I can do `var k string` though and drop the `:=` though it seems icky to use loop vars outside of the loop

Reply:
but that'd be incorrect for empty keys

Reply:
so in short n counts the number of succcesful writes, it happens to be len(keys) if there are no errors but can be shorter if there is an error.